drivers/spi/spi_ll_stm32.c

```diff
@@ -1,36 +1,171 @@
 /*
  * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2018 Bobby Noelte
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/* Keep first */
+#include "spi_ll_stm32.h"
+
 #define SYS_LOG_LEVEL CONFIG_SYS_LOG_SPI_LEVEL
 #include <logging/sys_log.h>
 
 #include <misc/util.h>
 #include <kernel.h>
-#include <board.h>
-#include <errno.h>
 #include <spi.h>
-#include <toolchain.h>
 
 #include <clock_control/stm32_clock_control.h>
 #include <clock_control.h>
 
-#include "spi_ll_stm32.h"
+#include "spi_context.h"
+
+/* Driver Settings
+ */
+#if CONFIG_HAS_DTS_SPI && defined(CONFIG_SOC_SERIES_STM32F0X)
```

Comment:
Is that possible to put this whole part in a specific file, to keep clean pure driver part?

Reply:
@erwango 
> Is that possible to put this whole part in a specific file,

Do you want two driver variants spi_ll_stm32 and a new one or do you want the driver settings to be transfered to another header file?

Reply:
I meant put drivers settings in a new header file.
I don't think that we should maintain two variants of the driver. One variant is a good amount of work already.

Reply:
My strategy is to have two driver variants during transition time. The new/enhanced driver is started as a pure copy of the current one to enable easy tracking of changes. After all driver users have switched to the new driver variant it should replace the current one and only one driver should be there to maintain.